compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -489,14 +489,20 @@ public void visit(BLangXMLNavigationAccess xmlNavigation, AnalyzerData data) {
     }
 
     private void checkXMLNamespacePrefixes(List<BLangXMLElementFilter> filters, AnalyzerData data) {
+        Map<Name, BXMLNSSymbol> nameBXMLNSSymbolMap = symResolver.resolveAllNamespaces(data.env);
+        BXMLNSSymbol defaultNSSymbol = nameBXMLNSSymbolMap.get(names.fromString(XMLConstants.DEFAULT_NS_PREFIX));
+
         for (BLangXMLElementFilter filter : filters) {
-            if (!filter.namespace.isEmpty()) {
-                Name nsName = names.fromString(filter.namespace);
-                BSymbol nsSymbol = symResolver.lookupSymbolInPrefixSpace(data.env, nsName);
-                filter.namespaceSymbol = nsSymbol;
-                if (nsSymbol.getKind() != SymbolKind.XMLNS) {
+            String namespace = filter.namespace;
+            if (!namespace.isEmpty()) {
+                Name nsName = names.fromString(namespace);
+                BXMLNSSymbol nsSymbol = nameBXMLNSSymbolMap.get(nsName);
+                if (nsSymbol == null) {
                     dlog.error(filter.nsPos, DiagnosticErrorCode.CANNOT_FIND_XML_NAMESPACE, nsName);
                 }
+                filter.namespaceSymbol = nsSymbol;
+            } else {
+                filter.namespaceSymbol = defaultNSSymbol;
```

Comment:
Shouldn't we set this only if `defaultNSSymbol` is not null?

Reply:
We only need the null check when there the namespace is not empty right? In the later Desugar stage when we have null checks to see if the namespaceSymbol is empty

Reply:
I meant 
```suggestion
            } else if (defaultNSSymbol != null) {
                filter.namespaceSymbol = defaultNSSymbol;
```

or better yet assign that check to something like `hasDefaultNS` and check and set here.

Reply:
Updated